docs/guides/intro-to-tests.md

```diff
@@ -0,0 +1,154 @@
+
+#### 2. Test setup
+
+```js
+const input = "hello";
+const actualResult = createWordCount(input);
+const expectedResult = {
+  hello: 1,
+};
+```
+
+In this section, we're declaring some important values that we will need to compare later on in the test.
+We declare an `input` variable that stores a value that we will pass to the function we are testing - `createWordCount`.
+We declare an `actualResult` variable that stores the **return value** of our function `createWordCount`.
+We declare an `expectedResult` which stores what we **expect** the `actualResult` to be when our function is behaving according to the test description.
+
+❗ Note: The test setup will vary from test to test - so you can expect to see different variable names altogether in different tests. The setup here is done in order to make it clearer what values will be compared later on in the test.
+
+#### 3. Assertion
+
+```js
+expect(actualResult).toEqual(expectedResult);
+```
+
+The assertion is the part of the test code that actually checks to see if something is true. In this example, we are claiming that the following is true:
+
+> "We expect `actualResult` **to equal** `expectedResult`"
+
+Notice that the statement above is very similar to the syntax used in the test code.
+
+The function [`toEqual`](https://jestjs.io/docs/expect#toequalvalue) ( which is part of the Jest framework ) is used to check that `actualResult` ( `createWordCount`'s return value ) and `expectedResult` are equal to each other. There are [many other functions](https://jestjs.io/docs/using-matchers) like `toEqual` which we can use to make different assertions.
+
+### Pass ✅ or Fail ❌
+
+Finally, our test can either pass or fail.
+
+- If `actualResult` is equal to `expectedResult` then the test has passed ✅.
+- If `actualResult` is _not_ equal to `expectedResult` then the test has failed ❌.
+
+## Understanding test feedback
+
+Consider the extract from the test file `example.test.js`
+
+![Test case for createWordCount](./assets/test-case.png)
+
+After the test has run ( we'll learn how to run the test code in a separate guide later on), we will get some feedback from Jest depending on whether the test passed or failed.
+If the test passes then we should expect to see something like this:
+
+![Passing test case for createWordCount](./assets/test-pass-case.png)
+
+If the test fails, then we should get some feedback that looks like this:
+
+![Failing test case for createWordCount](./assets/test-fails-case.png)
+
+Jest aims to give feedback that makes it simple to see why test has failed. It shows us several things:
+
+- the line number for the failing test (line 9)
+- shows us `expected` - which corresponds to our `expectedResult`
+- shows us `received` - which corresponds to our `actualResult`
+
+We can see the test feedback is telling us that `received` is `undefined`. In other words, `actualResult` ( `createWordCount`'s return value) is `undefined`.
+
+## Further reading
```

Comment:
Perhaps adding some suggested reading/watching would be good here?

I always enjoy Coding Train: https://www.youtube.com/watch?v=S3QwafQEvSs